.github/workflows/ci.yaml

```diff
@@ -49,20 +49,30 @@ jobs:
     steps:
     - uses: actions/checkout@v2
 
-    - name: Install Rust Toolchain Components
+    - name: Install rustfmt
       uses: actions-rs/toolchain@v1
       with:
-        components: clippy, rustfmt
+        components: rustfmt
         override: true
         toolchain: stable
 
+    - name: Install clippy nightly
+      uses: actions-rs/toolchain@v1
+      with:
+        components: clippy
+        toolchain: nightly
+
     - uses: Swatinem/rust-cache@v2
+      with:
+        key: "clippy-nightly"
```

Comment:
Hopefully this will mean that -- once we revert this PR -- we will not suffer from bigger caches anymore.